src/components/calcite-ui-icon/calcite-ui-icon.tsx

```diff
@@ -0,0 +1,182 @@
+import {
+  Component,
+  h,
+  Element,
+  Prop,
+  State,
+  Watch,
+  Host,
+  Build
+} from "@stencil/core";
+import { CSS } from "./resources";
+import { getElementDir } from "../../utils/dom";
+import { fetchIcon, scaleToPx } from "./utils";
+import { Scale } from "../../interfaces/UIIcon";
+import { Theme } from "../../interfaces/common";
+
+@Component({
+  assetsDirs: ["assets"],
+  tag: "calcite-ui-icon",
+  styleUrl: "calcite-ui-icon.scss",
+  shadow: true
+})
+export class CalciteUIIcon {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element()
+  el: HTMLElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * When true, the icon will be filled.
+   */
+  @Prop({
+    reflect: true
+  })
+  filled: boolean = false;
+
+  /**
+   * The UI icon to display. The value of this property must match the icon name from https://esri.github.io/calcite-ui-icons/.
+   */
+  @Prop({
+    reflect: true
+  })
+  icon: string = null;
+
+  /**
+   * When true, the icon will be mirrored when the element direction is 'rtl'.
+   */
+  @Prop({
+    reflect: true
+  })
+  mirrored: boolean = false;
+
+  /**
+   * Icon scale. Can be "s" | "m" | "l".
+   */
+  @Prop({
+    reflect: true
+  })
+  scale: Scale = "m";
+
+  /**
+   * Icon theme. Can be "light" or "dark".
+   */
+  @Prop({
+    reflect: true
+  })
+  theme: Theme = "light";
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  connectedCallback(): void {
+    this.waitUntilVisible(() => {
+      this.visible = true;
+      this.loadIconPathData();
+    });
+  }
+
+  disconnectedCallback(): void {
+    if (this.intersectionObserver) {
+      this.intersectionObserver.disconnect();
+      this.intersectionObserver = null;
+    }
+  }
+
+  async componentWillLoad(): Promise<void> {
+    this.loadIconPathData();
+  }
+
+  render() {
+    const { el, mirrored, pathData, scale } = this;
+    const dir = getElementDir(el);
+    const size = scaleToPx[scale];
+
+    return (
+      <Host role="img">
+        <svg
+          class={{
+            [CSS.icon]: true,
+            [CSS.mirrored]: dir === "rtl" && mirrored
+          }}
+          xmlns="http://www.w3.org/2000/svg"
+          height={size}
+          width={size}
+          viewBox={`0 0 ${size} ${size}`}
+        >
+          <path d={pathData} />
```

Comment:
We would put the title like this right?

```jsx
 {title ? <title>{title}</title> : null}
    <path d={path} />
```

Reply:
I'll create a follow-up issue for handling `title`/`aria-label`.